Headers/DebugServer2/Architecture/X86/RegisterCopy.h

```diff
@@ -0,0 +1,165 @@
+//
+// Copyright (c) 2014-present, Facebook, Inc.
+// All rights reserved.
+//
+// This source code is licensed under the University of Illinois/NCSA Open
+// Source License found in the LICENSE file in the root directory of this
+// source tree. An additional grant of patent rights can be found in the
+// PATENTS file in the same directory.
+//
+
+// Yes, I just decided to write macro vomit.
+
+#ifndef __DebugServer2_Architecture_X86_RegisterCopy_h
+#define __DebugServer2_Architecture_X86_RegisterCopy_h
+
+#include "DebugServer2/Architecture/CPUState.h"
+
+namespace ds2 {
+namespace Architecture {
+namespace X86 {
+
+// Macros are shit.
+#define _PASTE(A, B) A##B
+#define PASTE(A, B) _PASTE(A, B)
```

Comment:
Why is this necessary? (rather than just saying #define PASTE(A, B) A##B). I'll admit I'm not a macro pro